mpa/det/inferrer.py

```diff
@@ -161,11 +161,21 @@ def dummy_dump_features_hook(mod, inp, out):
 
         feature_vector_hook = dump_features_hook if dump_features else dummy_dump_features_hook
 
+        # Class-wise Saliency map for Single-Stage Detector, otherwise use class-ignore saliency map.
+        if not dump_saliency_map:
+            saliency_hook = nullcontext()
+        elif hasattr(model, 'bbox_head'):
+            saliency_hook = DetSaliencyMapHook(eval_model.module)
+        else:
+            saliency_hook = SaliencyMapHook(eval_model.module.backbone)
+
         # Use a single gpu for testing. Set in both mm_val_dataloader and eval_model
         if is_module_wrapper(model):
             model = model.module
+
+        # Inference with hooks
         with eval_model.module.backbone.register_forward_hook(feature_vector_hook):
-            with DetSaliencyMapHook(eval_model.module) if dump_saliency_map else nullcontext() as saliency_hook:
+            with saliency_hook:
```

Comment:
Could you explain why we could not resolve this issue in the previous PR -> https://github.com/openvinotoolkit/model_preparation_algorithm/pull/99/files ?

Reply:
This is my bad, should have fix this too